File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    
    def recursion(start, acc):
        nonlocal answer
        if start == len(numbers):
            if acc == target:
                answer += 1
            return
        
        recursion(start + 1, acc + numbers[start])
        recursion(start + 1, acc - numbers[start])

    recursion(0, 0)
    
    return answer

File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.js
// 해당 수를 더하거나/빼거나 두 가지를 분기로 처리

function solution(numbers, target) {    
    function recursion(sum, index) {
        if (index === numbers.length) {
            if (sum === target) answer += 1
            return
        }
        
        recursion(sum + numbers[index], index + 1)
        recursion(sum - numbers[index], index + 1)
    }
    
    let answer = 0;
    recursion(0, 0);
    return answer;
}